lib/compilers.js

```diff
@@ -342,7 +343,7 @@ function footnote(node) {
         'position': node.position
     });
 
-    return self.footnoteReference({
+    return self.visitors.footnoteReference({
```

Comment:
Your missing the context object for `footnoteReference` here, it should be invoked with the compiler (`self`), e.g. `...footnoteReference.call(this, ...`!